extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceCallback.kt

```diff
@@ -0,0 +1,78 @@
+package com.mapbox.maps.extension.androidauto
+
+import android.graphics.Rect
+import androidx.car.app.AppManager
+import androidx.car.app.CarContext
+import androidx.car.app.SurfaceCallback
+import androidx.car.app.SurfaceContainer
+import com.mapbox.common.Logger
+import com.mapbox.maps.MapInitOptions
+import com.mapbox.maps.MapboxExperimental
+
+/**
+ * @see MapboxCarMap to create new map experiences.
+ *
+ * This is a Mapbox implementation for the [SurfaceCallback]. It is used to simplify the lower
+ * level calls that manage the map surface. This class handles the surface callbacks and forwards
+ * them to the [CarMapSurfaceOwner] where [MapboxCarMapObserver] instances are notified.
+ */
+@OptIn(MapboxExperimental::class)
+internal class CarMapSurfaceCallback internal constructor(
+  private val carContext: CarContext,
+  private val carMapSurfaceOwner: CarMapSurfaceOwner,
+  private val mapInitOptions: MapInitOptions
```

Comment:
Is the carContext duplicated here? since MapInitOptions contains a context already?

Reply:
duplicate is a word for it. I was thinking of it as a cached reference. 

alternative is to cast it to CarContext every time it is needed. my preference is to leave it this way, do you have a strong opinion to cast it?

Reply:
I don't have a strong opinion on it, and I guess it doesn't really matter as the CarMapSurfaceCallback is an internal class.